riscv/processor.h

```diff
@@ -276,8 +276,37 @@ static int cto(reg_t val)
   return res;
 }
 
+class isa_parser_t {
+public:
+  isa_parser_t(): xlen(0), extension_table(256, false){};
```

Comment:
This is great.

Is there any reason the constructor can't take the ISA string as a parameter? Instead of having to call `parse_isa_string()` later to initialize it?